typescript/ts-converter/src/AstConverter.ts

```diff
@@ -865,6 +866,33 @@ export class AstConverter {
     return decl
   }
 
+  convertSwitchStatement(statement: ts.SwitchStatement): StatementDeclaration {
+    let expression = this.astExpressionConverter.convertExpression(statement.expression);
+    let cases: Array<CaseDeclarationProto> = [];
+    for (let clause of statement.caseBlock.clauses) {
+      let body: Array<StatementDeclaration> = [];
+      for (let statement of clause.statements) {
+        body = body.concat(this.convertStatement(statement));
+      }
+
+      if (ts.isCaseClause(clause)) {
```

Comment:
We can move if to the clause resolution here

Reply:
Can you clarify? I don't really understand this comment